code/game/machinery/big_manipulator.dm

```diff
@@ -0,0 +1,264 @@
+	SIGNAL_HANDLER
+
+	deconstruct(TRUE)
+
+/// Pre take and drop proc from [take and drop procs loop]:
+/// Check if we have item on take_turf to start take and drop loop
+/obj/machinery/big_manipulator/proc/is_work_check()
+	for(var/obj/item/take_item in take_turf.contents)
+		try_take_thing(take_turf, take_item)
+		break
+
+/// First take and drop proc from [take and drop procs loop]:
+/// Check if we can take item from take_turf to work with him. This proc also calling from ATOM_ENTERED signal.
+/obj/machinery/big_manipulator/proc/try_take_thing(datum/source, atom/movable/target)
+	SIGNAL_HANDLER
+
+	if(!on)
+		return
+	if(!anchored)
+		return
+	if(QDELETED(source) || QDELETED(target))
+		return
+	if(on_work)
+		return
+	if(!use_energy(active_power_usage, force = FALSE))
+		on = FALSE
+		say("Not enough energy!")
+		return
+	if(isitem(target))
+		start_work(target)
+
+/// Second take and drop proc from [take and drop procs loop]:
+/// Taking our item and start manipulator hand rotate animation.
+/obj/machinery/big_manipulator/proc/start_work(atom/movable/target)
+	target.forceMove(src)
+	containment_obj = WEAKREF(target)
+	on_work = TRUE
+	do_rotate_animation(1)
+	addtimer(CALLBACK(src, PROC_REF(drop_thing), target), working_speed)
+
+/// Third take and drop proc from [take and drop procs loop]:
+/// Drop our item and start manipulator hand backward animation.
+/obj/machinery/big_manipulator/proc/drop_thing(atom/movable/target)
+	target.forceMove(drop_turf)
+	do_rotate_animation(0)
+	addtimer(CALLBACK(src, PROC_REF(end_work)), working_speed)
+
+/// Fourth and last take and drop proc from take and drop procs loop:
+/// Finishes work and begins to look for a new item for [take and drop procs loop].
+/obj/machinery/big_manipulator/proc/end_work()
+	on_work = FALSE
+	is_work_check()
+
+/// Rotates manipulator hand 90 degrees.
+/obj/machinery/big_manipulator/proc/do_rotate_animation(backward)
+	animate(manipulator_hand, transform = matrix(90, MATRIX_ROTATE), working_speed*0.5)
+	addtimer(CALLBACK(src, PROC_REF(finish_rotate_animation), backward), working_speed*0.5)
+
+/// Rotates manipulator hand from 90 degrees to 180 or 0 if backward.
+/obj/machinery/big_manipulator/proc/finish_rotate_animation(backward)
+	animate(manipulator_hand, transform = matrix(180 * backward, MATRIX_ROTATE), working_speed*0.5)
+
+/obj/machinery/big_manipulator/ui_interact(mob/user, datum/tgui/ui)
+	if(!anchored)
+		to_chat(user, span_warning("[src] isn't attached to the ground!"))
+		return
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "BigManipulator")
+		ui.open()
+
+/obj/machinery/big_manipulator/ui_data(mob/user)
+	var/list/data = list()
+	data["active"] = on
+	return data
+
+/obj/machinery/big_manipulator/ui_act(action, params, datum/tgui/ui)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("on")
+			on = !on
+			if(on)
+				RegisterSignal(take_turf, COMSIG_ATOM_ENTERED, PROC_REF(try_take_thing))
```

Comment:
i'd turn this on off into a proc so you can map a version of this that is on by default

Reply:
Be cool if you can put the code to fast commit